src/components/auth/signIn/email/ConfirmationModalContent.tsx

```diff
@@ -0,0 +1,149 @@
+import { isStr } from '@subsocial/utils'
+import { Button, Form } from 'antd'
+import jwtDecode from 'jwt-decode'
+import { useState } from 'react'
+import AuthCode from 'react-auth-code-input'
+import {
+  confirmEmail,
+  getErrorMessage,
+  JwtPayload,
+  resendEmailConfirmation,
+} from 'src/components/utils/OffchainSigner/api/requests'
+import CountdownTimerButton from 'src/components/utils/OffchainSigner/CountdownTimerButton'
+import {
+  getSignerToken,
+  getTempRegisterAccount,
+} from 'src/components/utils/OffchainSigner/ExternalStorage'
+import { CODE_DIGIT } from 'src/config/ValidationsConfig'
+import useSignerExternalStorage from 'src/hooks/useSignerExternalStorage'
+import { StepsEnum } from '../../AuthContext'
+import styles from './SignInModalContent.module.sass'
+import { RegexValidations, useFormValidation } from './useFormValidation'
+
+type FormValues = {
+  confirmationCode: number
+}
+
+type FieldName = keyof FormValues
+
+export const fieldName = (name: FieldName): FieldName => name
+
+type Props = {
+  setCurrentStep: (step: number) => void
+}
+
+const ConfirmationModalContent = ({ setCurrentStep }: Props) => {
+  const { isValidCode } = useFormValidation()
+
+  const { setSignerTokensByAddress } = useSignerExternalStorage()
+
+  const userAddress = getTempRegisterAccount()
+
+  const [form] = Form.useForm()
+
+  const [isFormValid, setIsFormValid] = useState(false)
+  const [error, setError] = useState<string | null>(null)
+
+  const handleSubmit = async (values: FormValues) => {
+    const accessToken = getSignerToken(userAddress!)
+    if (!accessToken) throw new Error('Access token is not defined')
```

Comment:
same as above, shouldn't it be setError instead of throw?

Reply:
Applied in [6b8698e](https://github.com/dappforce/polkaverse/pull/74/commits/6b8698e777e002dabb46284e32ce9c7fac3c46e9)